### ISSUE TITLE
iflight bliz f722 v1.2 can't use i2c

### ISSUE THREAD

**Issue by MisakaRan** [iNavFlight/inav#10150]
## Current Behavior

﻿
## Steps to Reproduce

FC is Iflight bliz f722 v1.2 firmware is inav_7.1.2_IFLIGHT_BLITZ_F722_X1_OSD,I  connected the compass to SCL and SDA,FC Unable to detect compass
If this FC use inav_7.1.2_IFLIGHT_BLITZ_F722 ,it can be detected the compass,but FC can't detece the acc and gyro.
There are two versions of flight control，Iflight bliz f722 v1.2 use icm42688,Iflight bliz f722 v1.1 use bmi270
Iflight bliz f722 v1.1 use bmi270 firmware use IFLIGHT_BLITZ_F722 to work normally.
## Expected behavior

﻿FC should have detected the magnetic compass（If firmware IFLIGHT_BLITZ_F722_X1_OSD is designed for  Iflight bliz f722 v1.2 ）
## Suggested solution(s)

﻿Modify the firmware of  Iflight bliz f722 v1.2.
## Additional context

﻿# status 
INAV/IFLIGHT_BLITZ_F722_X1_OSD 7.1.2 May 31 2024 / 22:55:42 (4e1e59eb)
GCC-10.3.1 20210824 (release)
System Uptime: 10 seconds
Current Time: 2041-06-28T01:04:00.000+00:00
Voltage: 0.00V (0S battery - NOT PRESENT)
CPU Clock=216MHz, GYRO=ICM42605, ACC=ICM42605, BARO=DPS310, RANGEFINDER=MSP, OPFLOW=MSP
STM32 system clocks:
  SYSCLK = 216 MHz
  HCLK   = 216 MHz
  PCLK1  = 54 MHz
  PCLK2  = 108 MHz
Sensor status: GYRO=OK, ACC=OK, MAG=NONE, BARO=OK, RANGEFINDER=FAILING, OPFLOW=FAILING, GPS=OK
ESC Temperature(s): Motor Count = 4
ESC 0: 0°C, ESC 1: 0°C, ESC 2: 0°C, ESC 3: 0°C, 
Stack size: 6144, Stack address: 0x20010000, Heap available: 1820
I2C Errors: 50, config size: 10455, max available config: 16384
ADC channel usage:
   BATTERY : configured = ADC 1, used = ADC 1
      RSSI : configured = none, used = none
   CURRENT : configured = ADC 2, used = ADC 2
  AIRSPEED : configured = none, used = none
System load: 26, cycle time: 540, PID rate: 1851, RX rate: 9, System rate: 9
Arming disabled flags: NAV HWFAIL RX CLI DSHOTBEEPER
OSD: MSP DisplayPort [53 x 20]
VTX: not detected
GPS: HW Version: UBLOX9 Proto: 32.00 Baud: 115200
  GNSS Capabilities:
    GNSS Provider active/default
    GPS 1/1
    Galileo 1/1
    BeiDou 0/1
    Glonass 0/1
    Max concurrent: 4

---

# version 
# INAV/IFLIGHT_BLITZ_F722_X1_OSD 7.1.2 May 31 2024 / 22:55:42 (4e1e59eb)
# GCC-10.3.1 20210824 (release)

﻿
- FC Board name and vendor:
- INAV version string:
﻿

**Comment by dpyeates** [comment]
Hi,

I'm having similar issues.

_"If this FC use inav_7.1.2_IFLIGHT_BLITZ_F722 ,it can be detected the compass,but FC can't detece the acc and gyro"_
Same for me. No Acc or Gyro if I use the IFLIGHT_BLITZ_F722 firmware.

Oh! I just tried flashing the IFLIGHT_BLITZ_F722_X1_OSD firmware to my IFLIGHT_BLITZ_F722 and the Acc and Gyro work!

**Comment by dpyeates** [comment]
So I did a reset to the 7.1.2 release (git reset --hard ed7fb3b) and took a look at the src/main/target/IFLIGHT_BLITZ_F722/target.h.

Turns out, that the following code, required to make the Acc and Gyro work is not present!

// ICM42605
#define USE_IMU_ICM42605
#define IMU_ICM42605_ALIGN      CW0_DEG
#define ICM42605_SPI_BUS        BUS_SPI1
#define ICM42605_CS_PIN         PA4

So, the 7.1.2 build of the IFLIGHT_BLITZ_F722 will not work with the ICM42605 series 

I added the above code back in and then rebuilt the target. The new build is attached. You will need to rename it to a *.hex file (its not a zip file so don't try and unzip it!)

[inav_7.1.2_IFLIGHT_BLITZ_F722.zip](https://github.com/user-attachments/files/15930810/inav_7.1.2_IFLIGHT_BLITZ_F722.zip)

**Comment by MisakaRan** [comment]
@dpyeates it worked normally,i ll try it tomorrow

**Comment by mmosca** [comment]
Can you guys post a picture of your FCs?

It is likely the case of a mistaken identity target.

**Comment by dpyeates** [comment]
Here you go. An iFlight Blitz F7. I'm using the right target.

![IMG_2243](https://github.com/iNavFlight/inav/assets/944670/5e883720-80fa-46d2-b9bf-70e29b5aaf7d)

The issue is that 7.1.2 build (from tag ed7fb3b) does NOT include the required defines in target.h. If you do a git hard reset to this you will see. The _current_ version of the IFLIGHT_BLITZ_F722/target.h DOES have this included, so I'm not sure why there is a discrepancy.

Bottom Line, when I added the above defines to the IFLIGHT_BLITZ_F722/target.h and did a local rebuild, and flashed the local firmware, the acc/gyro works normally.

**Comment by mmosca** [comment]
> Here you go. An iFlight Blitz F7. I'm using the right target.
> 

Try IFLIGHT_BLITZ_F722 target on INAV's 8 nightly builds. The gyro has been added and will be in the next major release.
